protocol/x/prices/keeper/market.go

```diff
@@ -21,6 +21,14 @@ func (k Keeper) CreateMarket(
 	marketParam types.MarketParam,
 	marketPrice types.MarketPrice,
 ) (types.MarketParam, error) {
+	if _, exists := k.GetMarketParam(ctx, marketParam.Id); exists {
```

Comment:
Updated signature of `GetMarketParam` to be more generalized (before it throws an error if `id` is not found)